src/main/java/dev/blaauwendraad/masker/json/KeyMatcher.java

```diff
@@ -249,11 +250,20 @@ static class TrieNode {
         /**
          * Masking configuration for the key that ends at this node.
          */
-        @CheckForNull
+        @Nullable
         KeyMaskingConfig keyMaskingConfig = null;
         /**
          * Used to store the configuration, but indicate that json-masker is in ALLOW mode and the key is not allowed.
          */
         boolean negativeMatch = false;
+
+        @Nullable
+        public TrieNode child(byte b) {
+            return children[b + BYTE_OFFSET];
+        }
+
+        public void add(byte b, TrieNode child) {
+            children[b + BYTE_OFFSET] = child;
+        }
```

Comment:
I don't really like that these methods don't have JavaDoc. The whole KeyMatcher is already complex enough including good docs (at least for me) 😅

Reply:
Especially since they are public (do they need to be?)

Reply:
The class itself is package private, so I could make them that as well, I didn't put too much thought into it just got tired of messing with the offsets 😅

Reply:
Yea, let's make them package private and add some docs